LEGO1/lego/legoomni/src/infocenter/infocenter.cpp

```diff
@@ -46,10 +47,29 @@ Infocenter::Infocenter()
 	m_unk0x1d6 = 0;
 }
 
-// STUB: LEGO1 0x1006ec90
+// FUNCTION: LEGO1 0x1006ec90
 Infocenter::~Infocenter()
 {
-	// TODO
+	BackgroundAudioManager()->Stop();
+
+	MxS16 i = 0;
+	do {
```

Comment:
Did you try if a regular for-loop matches as well? do/while loops are relatively rare